flink-yarn/src/main/java/org/apache/flink/yarn/Utils.java

```diff
@@ -596,6 +644,30 @@ static ContainerLaunchContext createTaskExecutorContext(
 		return ctx;
 	}
 
+	static Map<String, FileStatus> getPreUploadedFlinkFiles(
+		@Nullable String flinkSharedBinary,
+		Configuration yarnConfiguration) {
+		Map<String, FileStatus> flinkShareFiles = new HashMap<>();
+		if (flinkSharedBinary != null) {
+			try {
+				FileSystem fileSystem = FileSystem.get(yarnConfiguration);
+				Path flinkSharedPath = new Path(flinkSharedBinary);
+				if (fileSystem.exists(flinkSharedPath)) {
+					RemoteIterator<LocatedFileStatus> iterable = fileSystem.listFiles(flinkSharedPath, true);
+					while (iterable.hasNext()) {
+						LocatedFileStatus locatedFileStatus = iterable.next();
+						String relative = flinkSharedPath.toUri().relativize(locatedFileStatus.getPath().toUri()).getPath();
+						flinkShareFiles.put(relative, locatedFileStatus);
+						LOG.debug("Found pre-uploaded file {} under {}", relative, flinkSharedPath);
+					}
+				}
+			} catch (IOException e) {
+				LOG.warn("Error to get the ");
```

Comment:
@walterddr @wangyang0918 please fix this log when merging.

Reply:
oops. missed this one.
```
LOG.warn("Error getting pre-uploaded Flink binary via path: {}", flinkSharedBinary, e);
```

Reply:
Nice catch. I will fix it.